src/WAQI.php

```diff
@@ -117,6 +117,37 @@ public function getObservationByStation(string $station = 'here'): void
         }
     }
 
+    public function getObservationByGeoLocation(float $latitude, float $longitude): void
+    {
+        $client = new Client(['base_uri' => self::API_ENDPOINT]);
+
+        try {
+            $response = $client->request('GET', 'feed/geo:' . $latitude . ';' . $longitude . '/', ['query' => 'token='.$this->token]);
+        } catch (ClientException $e) {
+            echo Psr7\str($e->getRequest());
+            echo Psr7\str($e->getResponse());
+            exit();
+        } catch (RequestException $e) {
+            echo Psr7\str($e->getRequest());
+            if ($e->hasResponse()) {
+                echo Psr7\str($e->getResponse());
+            }
+            exit();
+        } catch (GuzzleException $e) {
+            echo $e->getMessage();
+            exit();
+        }
+
+        $_response_body = \json_decode($response->getBody());
```

Comment:
Be aware that `$response->getBody()` returns a **StreamInterface** instance and `json_decode` expects a string value. It is better to wrap the response with this function: `Psr7\copy_to_string`